lib/extensions/ar_adapter.rb

```diff
@@ -1,3 +1,2 @@
-$:.push("#{File.dirname(__FILE__)}/ar_adapter")
 require "active_record/connection_adapters/postgresql_adapter"
-Dir.glob(File.expand_path(File.join(File.dirname(__FILE__), "ar_adapter", "*.rb"))) { |f| require File.basename(f, ".*") }
+Dir.glob(File.expand_path(File.join(__dir__, "ar_adapter", "*.rb"))).sort.each { |f| require f }
```

Comment:
I find this to be one of the most frustrating bugs...part of me just wants it fixed in Dir.glob, but I can understand where Ruby core might not like a fix there because it would introduce a sorting performance penalty, possibly.  Then again, I can't imagine where that penalty would be significant...I mean who Dir.globs a directory over and over in a loop?  ¯\\\_(ツ)_/¯  cc @tenderlove